microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/java/hover/PropertiesHoverParticipant.java

```diff
@@ -145,34 +142,16 @@ public Hover collectHover(JavaHoverContext context, IProgressMonitor monitor) th
 			return null;
 		}
 
-		String annotationSource = ((ISourceReference) annotation).getSource();
-		String propertyKey = null;
-		Range propertyKeyRange = null;
-		boolean found = false;
-		for (String annotationMemberName : annotationMembers) {
-			propertyKey = getAnnotationMemberValue(annotation, annotationMemberName);
-			if (propertyKey != null) {
-				ISourceRange r = ((ISourceReference) annotation).getSourceRange();
-				Pattern memberPattern = Pattern.compile(".*[^\"]\\s*(" + annotationMemberName + ")\\s*=.*",
-						Pattern.DOTALL);
-				Matcher match = memberPattern.matcher(annotationSource);
-				if (match.matches()) {
-					int offset = annotationSource.indexOf(propertyKey, match.end(1));
-					propertyKeyRange = utils.toRange(typeRoot, r.getOffset() + offset, propertyKey.length());
-
-					if (!hoverPosition.equals(propertyKeyRange.getEnd())
-							&& Ranges.containsPosition(propertyKeyRange, hoverPosition)) {
-						found = true;
-						break;
-					}
-				}
-			}
-		}
+		AnnotationMemberInfo annotationMemberInfo = getAnnotationMemberAt(annotation, annotationMembers, hoverPosition,
+				typeRoot, utils, false);
 
-		if (!found) {
+		String annotationMemberValue = annotationMemberInfo.getMemberValue();
+		if (annotationMemberValue == null) {
 			return null;
 		}
 
+		Range propertyKeyRange = annotationMemberInfo.getRange();
+

```

Comment:
Remove test with 

`if (javaProject == null) {`

I'm not sure this usecase exists.

Reply:
This seems dangerous to remove, are we sure?